src/ts/components/subtitlesettings/subtitlesetting.ts

```diff
@@ -0,0 +1,7 @@
+import {SubtitleOverlay} from '../subtitleoverlay';
+import {ListSelectorConfig} from '../listselector';
+
+
+export interface SubtitleSettingConfig extends ListSelectorConfig {
```

Comment:
Rename to `SubtitleSettingSelectBoxConfig`, implement the `SubtitleSettingSelectBox` in this file and rename the file to `subtitlesettingselectbox.ts`. Then extend all your select boxes from this class.

The general pattern is that each component gets a separate file, the file is named after the component in lower case, and the component can have an optional config interface named after the component plus "Config" suffix (without a config interface, the component uses the config interface of the parent class).